tests/utils/managed_process.py

```diff
@@ -111,15 +113,27 @@ def __enter__(self):
             raise e
 
     def __exit__(self, exc_type, exc_val, exc_tb):
-        process_list = [self.proc, self._tee_proc, self._sed_proc]
-        for process in process_list:
-            if process:
-                if process.stdout:
-                    process.stdout.close()
-                if process.stdin:
-                    process.stdin.close()
-                terminate_process_tree(process.pid, self._logger)
-                process.wait()
+        # First, terminate the process to stop output generation
+        if self._proc:
+            terminate_process_tree(self._proc.pid, self._logger)
+            self._proc.wait()
+
+        # Wait for output processing thread to finish
+        if (
+            hasattr(self, "_output_thread")
```

Comment:
Isn't this always true by definition of the class? If the attribute is missing then this should rightfully throw an error and the class should be fixed